chain/syncer.go

```diff
@@ -188,7 +187,14 @@ func (syncer *Syncer) collectChain(ctx context.Context, tsKey types.TipSetKey) (
 	// Continue collecting the chain if we're either not yet caught up or the
 	// number of new input blocks is less than the FinalityLimit constant.
 	// Otherwise, halt assuming the new blocks come from an invalid chain.
-	for (syncer.syncMode == Syncing) || !syncer.exceedsFinalityLimit(chain) {
+	for {
+		exceedsFinality, err := syncer.exceedsFinalityLimit(chain)
+		if err != nil {
+			return nil, err
+		}
+		if !(syncer.syncMode == Syncing) && exceedsFinality {
```

Comment:
Simplify the first test to `syncMode != Syncing`

Reply:
whoops thanks